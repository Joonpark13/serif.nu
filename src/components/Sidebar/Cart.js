import React from 'react';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';

export const styles = {
  cartHeading: {
    padding: '16px',
  },
};

function Cart({ classes, sections }) {
  const label = sections.length === 1 ? 'Class' : 'Classes';
  return (
    <Typography variant="h5" className={classes.cartHeading}>
      {sections.length}
      {' '}
      {label}
    </Typography>
  );
}

Cart.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  sections: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export { Cart as UnstyledCart };
export default withStyles(styles)(Cart);
