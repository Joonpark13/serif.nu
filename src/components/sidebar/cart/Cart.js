import React from 'react';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/styles';
import PropTypes from 'prop-types';
import CartSection from './CartSection';

export const styles = {
  cartHeading: {
    padding: '16px',
  },
};

function Cart({ classes, sections }) {
  const uniqueSections = [];
  const usedIDs = [];

  sections.forEach(
    (section) => {
      if (!usedIDs.includes(section.id)) {
        usedIDs.push(section.id);
        uniqueSections.push(section);
      }
    },
  );

  const label = uniqueSections.length === 1 ? 'Class' : 'Classes';

  return (
    <React.Fragment>
      <Typography variant="h5" className={classes.cartHeading}>
        {uniqueSections.length}
        {' '}
        {label}
      </Typography>
      {uniqueSections.map(section => <CartSection key={section.id} section={section} />)}
    </React.Fragment>
  );
}

Cart.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  sections: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export { Cart as UnstyledCart };
export default withStyles(styles)(Cart);
