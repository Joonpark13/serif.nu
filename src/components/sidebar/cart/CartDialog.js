import React, { useState } from 'react';
import { Dialog, DialogActions, DialogTitle, Button } from '@material-ui/core';
import { useDispatch } from 'react-redux';
import { removeAllClasses } from 'actions';

export default function CartDialog() {
  const [open, setOpen] = useState(true);
  const dispatch = useDispatch();

  const handleRemoveAll = () => {
    dispatch(removeAllClasses());
    setOpen(false);
  };
  const handleClose = () => {
    setOpen(false);
  };

  return (
    <Dialog
      open={open}
      onClose={handleClose}
    >
      <DialogTitle>Are you sure you want to remove all classes?</DialogTitle>
      <DialogActions>
        <Button onClick={handleRemoveAll}>
            Remove
        </Button>
        <Button onClick={handleClose}>
            Cancel
        </Button>
      </DialogActions>
    </Dialog>
  );
}
