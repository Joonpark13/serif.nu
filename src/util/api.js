/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
