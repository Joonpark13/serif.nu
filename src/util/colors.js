export const northwesternPurple = '#4E2A84';
export const northwesternBrightOrange = '#FFC520';
export const northwesternRichBlack10 = '#D8D6D6';
