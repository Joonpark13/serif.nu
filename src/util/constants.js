/* eslint-disable import/prefer-default-export */
export const CURRENT_TERM = '4730';
