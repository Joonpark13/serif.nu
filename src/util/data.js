import termsData from 'data/terms.json';

/* eslint-disable import/prefer-default-export */
/* istanbul ignore next */
export const currentTerm = termsData.reduce((highestTerm, term) => {
  if (term.id > highestTerm.id) {
    return term;
  }
  return highestTerm;
});
/* eslint-enable import/prefer-default-export */
