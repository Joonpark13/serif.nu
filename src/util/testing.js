/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
