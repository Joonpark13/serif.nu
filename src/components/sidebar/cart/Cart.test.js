import React from 'react';
import { shallow } from 'enzyme';
import { mockUseSelector, testSection } from 'util/testing';
import { Button } from '@material-ui/core';
import Cart from './Cart';
import CartDialog from './CartDialog';

describe('Cart', () => {
  beforeEach(() => {
    mockUseSelector([]);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Cart />);

    expect(wrapper.get(0)).toMatchSnapshot();
  });

  it('renders correctly with one section', () => {
    mockUseSelector([testSection]);

    const wrapper = shallow(<Cart />);

    expect(wrapper.get(0)).toMatchSnapshot();
  });

  it('renders correctly with multiple sections with the same id', () => {
    mockUseSelector([testSection, testSection]);

    const wrapper = shallow(<Cart />);

    expect(wrapper.get(0)).toMatchSnapshot();
  });

  it('opens the modal when clicked', () => {
    mockUseSelector([testSection, testSection]);

    const wrapper = shallow(<Cart />);
    wrapper.find(Button).first().simulate('click');

    expect(wrapper.find(CartDialog)).toHaveLength(1);
  });
});
