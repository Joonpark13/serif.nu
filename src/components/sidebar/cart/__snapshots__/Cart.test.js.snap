// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart renders correctly 1`] = `
<React.Fragment>
  <ForwardRef(WithStyles)
    container={true}
    direction="row"
    justify="space-between"
  >
    <ForwardRef(WithStyles)
      className="makeStyles-cartHeading-1"
      variant="h5"
    >
      0
       
      Classes
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
</React.Fragment>
`;

exports[`Cart renders correctly with multiple sections with the same id 1`] = `
<React.Fragment>
  <ForwardRef(WithStyles)
    container={true}
    direction="row"
    justify="space-between"
  >
    <ForwardRef(WithStyles)
      className="makeStyles-cartHeading-1"
      variant="h5"
    >
      1
       
      Class
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      onClick={[Function]}
    >
       Remove All 
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
  <CartSection
    section={
      Object {
        "courseId": "101-1",
        "descriptions": Array [
          Object {
            "name": "",
            "value": "",
          },
        ],
        "id": "12345",
        "instructors": Array [
          "",
        ],
        "name": "Introduction to Something",
        "schedules": Array [
          Object {
            "dow": Array [
              "Mo",
            ],
            "end": Object {
              "hour": 12,
              "minute": 0,
            },
            "location": "Somewhere",
            "start": Object {
              "hour": 10,
              "minute": 30,
            },
          },
        ],
        "schoolId": "MEAS",
        "sectionNumber": "1",
        "subjectId": "COMP_SCI",
        "termId": "111111",
        "topic": "Section topic...",
      }
    }
  />
</React.Fragment>
`;

exports[`Cart renders correctly with one section 1`] = `
<React.Fragment>
  <ForwardRef(WithStyles)
    container={true}
    direction="row"
    justify="space-between"
  >
    <ForwardRef(WithStyles)
      className="makeStyles-cartHeading-1"
      variant="h5"
    >
      1
       
      Class
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      onClick={[Function]}
    >
       Remove All 
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
  <CartSection
    section={
      Object {
        "courseId": "101-1",
        "descriptions": Array [
          Object {
            "name": "",
            "value": "",
          },
        ],
        "id": "12345",
        "instructors": Array [
          "",
        ],
        "name": "Introduction to Something",
        "schedules": Array [
          Object {
            "dow": Array [
              "Mo",
            ],
            "end": Object {
              "hour": 12,
              "minute": 0,
            },
            "location": "Somewhere",
            "start": Object {
              "hour": 10,
              "minute": 30,
            },
          },
        ],
        "schoolId": "MEAS",
        "sectionNumber": "1",
        "subjectId": "COMP_SCI",
        "termId": "111111",
        "topic": "Section topic...",
      }
    }
  />
</React.Fragment>
`;
